src/web/workers/InputWorker.mjs

```diff
@@ -0,0 +1,1048 @@
+    const silent = inputData.silent;
+    const input = self.getInputObj(inputNum);
+    if (input === undefined || input === null) return;
+
+    let inputVal = input.data;
+    const inputObj = {
+        inputNum: inputNum,
+        input: inputVal
+    };
+    if (typeof inputVal !== "string") {
+        inputObj.name = inputVal.name;
+        inputObj.size = inputVal.size;
+        inputObj.type = inputVal.type;
+        inputObj.progress = input.progress;
+        inputObj.status = input.status;
+        inputVal = inputVal.fileBuffer;
+        const fileSlice = inputVal.slice(0, 512001);
+        inputObj.input = fileSlice;
+
+        self.postMessage({
+            action: "setInput",
+            data: {
+                inputObj: inputObj,
+                silent: silent
+            }
+        }, [fileSlice]);
+    } else {
+        self.postMessage({
+            action: "setInput",
+            data: {
+                inputObj: inputObj,
+                silent: silent
+            }
+        });
+    }
+    self.updateTabHeader(inputNum);
+};
+
+/**
+ * Gets the nearby inputNums to the provided number, and posts them
+ * to the inputWaiter to be displayed on the page.
+ *
+ * @param {number} inputNum - The inputNum to find the nearby numbers for
+ * @param {string} direction - The direction to search for inputNums in. Either "left" or "right"
+ */
+self.refreshTabs = function(inputNum, direction) {
+    const nums = self.getNearbyNums(inputNum, direction),
+        inputNums = Object.keys(self.inputs),
+        tabsLeft = (self.getSmallestInputNum(inputNums) !== nums[0]),
+        tabsRight = (self.getLargestInputNum(inputNums) !== nums[nums.length - 1]);
+
+    self.postMessage({
+        action: "refreshTabs",
+        data: {
+            nums: nums,
+            activeTab: (nums.includes(inputNum)) ? inputNum : self.getNextInputNum(inputNum),
+            tabsLeft: tabsLeft,
+            tabsRight: tabsRight
+        }
+    });
+
+    // Update the tab headers for the new tabs
+    for (let i = 0; i < nums.length; i++) {
+        self.updateTabHeader(nums[i]);
+    }
+};
+
+/**
+ * Update the stored status for an input
+ *
+ * @param {number} inputNum - The input that's having its status changed
+ * @param {string} status - The status of the input
+ */
+self.updateInputStatus = function(inputNum, status) {
+    if (self.inputs[inputNum] !== undefined) {
+        self.inputs[inputNum].status = status;
+    }
+};
+
+/**
+ * Update the stored load progress of an input
+ *
+ * @param {object} inputData
+ * @param {number} inputData.inputNum - The input that's having its progress updated
+ * @param {number} inputData.progress - The load progress of the input
+ */
+self.updateInputProgress = function(inputData) {
+    const inputNum = inputData.inputNum;
+    const progress = inputData.progress;
+
+    if (self.inputs[inputNum] !== undefined) {
+        self.inputs[inputNum].progress = progress;
```

Comment:
We might want to consider adding `@babel/plugin-proposal-optional-chaining` as it would allow us to use the following syntax:

```javascript
self.inputs[inputNum]?.progress = progress;
```
May not be worth it for this PR, but I'd like to use it in future. https://github.com/tc39/proposal-optional-chaining